sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs

```diff
@@ -331,14 +391,19 @@ protected ServiceBusSender()
         /// </summary>
         /// <param name="sequenceNumber">The <see cref="ServiceBusReceivedMessage.SequenceNumber"/> of the message to be cancelled.</param>
         /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
-        public virtual async Task CancelScheduledMessageAsync(
+        public virtual async Task CancelScheduledAsync(
```

Comment:
Has this been discussed? Not a big fan of this renaming. It reads awkwardly.

Reply:
We are discussing PeekBatch/ReceiveBatch interplay with the ServiceBusMessageBatch as well. The idea there was to instead use PeekMessage/PeekMessages to avoid overloading "batch", so this might be changing again. I will back out this change for now.

Reply:
You know my position on the word 'batch' - it has a strong meaning and it's not aligned with the receive behaviour of the broker. You're almost never given an exact number you're asking for.

> We are discussing PeekBatch/ReceiveBatch interplay with the ServiceBusMessageBatch as well. The idea there was to instead use PeekMessage/PeekMessages to avoid overloading "batch", so this might be changing again. I will back out this change for now.

Did you mean "PeekMessages/ReceiveMessages"? If yes, I'll agree. Those are much better names.

> I will back out this change for now.

👍

Reply:
> You know my position on the word 'batch' - it has a strong meaning and it's not aligned with the receive behaviour of the broker. You're almost never given an exact number you're asking for.
> 
> > We are discussing PeekBatch/ReceiveBatch interplay with the ServiceBusMessageBatch as well. The idea there was to instead use PeekMessage/PeekMessages to avoid overloading "batch", so this might be changing again. I will back out this change for now.
> 
> Did you mean "PeekMessages/ReceiveMessages"? If yes, I'll agree. Those are much better names.
> 
> > I will back out this change for now.
> 
> 👍

Yes, PeekMessages/ReceiveMessages, but this would compel us to also update the singular forms to PeekMessage/ReceiveMessage. We were trying to avoid having the single letter differentiation between methods, but I think along with the different return type it should be fairly distinguishable in Intellisense.

Reply:
> Yes, PeekMessages/ReceiveMessages, but this would compel us to also update the singular forms to PeekMessage/ReceiveMessage. We were trying to avoid having the single letter differentiation between methods, but I think along with the different return type it should be fairly distinguishable in Intellisense.

I'd rather have two methods than continuously provide a count of how many messages to retrieve when only one message is needed. Absolutely nothing's wrong with an explicit API where it makes sense. `ReceiveMessage` and `ReceiveMessageS` have their place under the sun.

Reply:
The other piece to this is that we would still likely just have SendAsync with various overloads. Since the return type is the same, we wouldn't need SendMessage/SendMessages/SendMessageBatch so this does introduce a slight inconsistency with the receive methods.

Reply:
True, but how bad it is to have those additional methods? When I contrast the options

Sending | Receiving
--|--
SendMessage | ReceiveMessage
SendMessages| ReceiveMessages
SendMessageBatch| ReceiveMessageBatch

vs

Sending | Receiving
--|--
Send| ReceiveMessage
Send| ReceiveMessages
Send| ReceiveMessageBatch

Option 1 is more appealing as there's no mental overload and it's very crystal clear what it does by just looking at the method name.

Reply:
Well we wouldn't have `ReceiveMessageBatch` anymore so it would be:
Sending | Receiving
--|--
SendMessage | ReceiveMessage
SendMessages| ReceiveMessages
SendMessageBatch| 

vs
Sending | Receiving
--|--
Send | ReceiveMessage
Send| ReceiveMessages
Send|